components/eventForm/LocationBanner.tsx

```diff
@@ -0,0 +1,102 @@
+import { useReverseGeocodeQuery } from "../../hooks/Geocoding"
+import { placemarkToFormattedAddress } from "../../lib/location"
+import React from "react"
+import { StyleSheet, View } from "react-native"
+import {
+  useEventFormContext
+} from "./EventForm"
+import { FormLabel, SkeletonFormLabel } from "../formComponents/FormLabels"
+import { MaterialIcons } from "@expo/vector-icons"
+import { FontScaleFactors, useFontScale } from "../../lib/FontScale"
+import { EventFormLocationInfo, EventFormPlacemarkInfo } from "./EventFormValues"
+
+/**
+ * Displays the selected location (if one) in the event form.
+ */
+export const EventFormLocationBanner = () => {
+  const locationInfo = useEventFormContext().watch("locationInfo")
+  return (
+    <View style={styles.bannerContainer}>
+      {!locationInfo
+        ? (
+          <FormLabel
+            style={styles.label}
+            icon="location-pin"
+            headerText="No Location Selected"
+            captionText="You must select a location to save this event."
+          />
+        )
+        : (
+          <LocationInfoBanner {...locationInfo} />
+        )}
+      <MaterialIcons
+        name="chevron-right"
+        size={
+          24 * useFontScale({ maximumScaleFactor: FontScaleFactors.xxxLarge })
+        }
+        color="black"
+        style={styles.bannerNavigationIcon}
+      />
+    </View>
+  )
+}
+
+const LocationInfoBanner = (locationInfo: EventFormLocationInfo) => {
+  return locationInfo.placemarkInfo
+    ? (
+      <PlacemarkInfoBanner {...locationInfo.placemarkInfo} />
+    )
+    : (
+      <GeocodedLocationInfoBanner {...locationInfo} />
+    )
+}
+
+const GeocodedLocationInfoBanner = (locationInfo: EventFormLocationInfo) => {
+  const { data: placemark, status } = useReverseGeocodeQuery(
```

Comment:
Could we combine these states in the GeocodedLocationInfoBanner, something like const placemark = placemarkFromProps ?? placemarkFromQuery, if placemark === null show the skeleton loader.

Reply:
I could possibly do something with disabling the query in react query to make that happen